src/main/java/duke/command/DukeCommand.java

```diff
@@ -0,0 +1,248 @@
+package duke.command;
+
+import duke.Storage;
+import duke.Ui;
+import duke.exception.InvalidCommandException;
+import duke.task.DukeDeadlineTask;
+import duke.task.DukeEvent;
+import duke.task.DukeSimpleTask;
+import duke.task.DukeTask;
+import duke.task.TaskList;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public enum DukeCommand implements DukeCommandAction {
```

Comment:
I believe I desperately need some comments here over what is happening. Would you kindly help me with that by adding in some comments?

Reply:
I agree with this review I also need some comments to understand what is happening